### PR TITLE
EPMDJ-5501 Removed useless engine and class data from Junit5

### PR DESCRIPTION
**Pull request by Mazel-Tovr**

### CODE REVIEW

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/junit/JUnit5Strategy.kt [hunk 1]
```diff
@@ -46,7 +46,9 @@ object JUnit5Strategy : AbstractTestStrategy() {
                 """
                             public void executionSkipped(org.junit.platform.engine.TestDescriptor testDescriptor, String reason) {
                                 mainRunner.executionSkipped(testDescriptor, reason);
-                                ${TestListener::class.java.name}.INSTANCE.${TestListener::testIgnored.name}(testDescriptor.getUniqueId().toString());
+                                if(testDescriptor.getType() == org.junit.platform.engine.TestDescriptor.Type.TEST) {
```
Comment (LikhanovMaxim) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/junit/JUnit5Strategy.kt:49]: formatting
```suggestion
                                if (testDescriptor.getType() == org.junit.platform.engine.TestDescriptor.Type.TEST) {
```

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/junit/JUnit5Strategy.kt [hunk 2]
```diff
@@ -68,7 +72,9 @@ object JUnit5Strategy : AbstractTestStrategy() {
                 """
                             public void executionFinished(org.junit.platform.engine.TestDescriptor testDescriptor, org.junit.platform.engine.TestExecutionResult testExecutionResult) {
                                 mainRunner.executionFinished(testDescriptor, testExecutionResult);
-                                ${TestListener::class.java.name}.INSTANCE.${TestListener::testFinished.name}(testDescriptor.getUniqueId().toString(), testExecutionResult.getStatus().name());
+                                if(testDescriptor.getType() == org.junit.platform.engine.TestDescriptor.Type.TEST) {
```
Comment (LikhanovMaxim) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/junit/JUnit5Strategy.kt:75]: this condition duplicates three times - fix

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/junit/JUnit5Strategy.kt [hunk 3]
```diff
@@ -68,7 +72,9 @@ object JUnit5Strategy : AbstractTestStrategy() {
...
+                                    ${TestListener::class.java.name}.INSTANCE.${TestListener::testFinished.name}(testDescriptor.getUniqueId().toString(), testExecutionResult.getStatus().name());
```
Comment (LikhanovMaxim) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/junit/JUnit5Strategy.kt:76]: testDescriptor.getUniqueId().toString() duplicates three times - maybe in const or variable

**Review by LikhanovMaxim — CHANGES_REQUESTED** [review]

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/junit/JUnit5Strategy.kt (hunk 2 above)
Comment (Mazel-Tovr) [reply src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/junit/JUnit5Strategy.kt:75]: are you suggesting something like this ? 
public boolean isTest(org.junit.platform.engine.TestDescriptor testDescriptor) {
  return testDescriptor.getType() == org.junit.platform.engine.TestDescriptor.Type.TEST;
}

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/junit/JUnit5Strategy.kt (hunk 3 above)
Comment (Mazel-Tovr) [reply src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/testing/junit/JUnit5Strategy.kt:76]: I don't think it's possible to put this in a constant

**Review by LikhanovMaxim — APPROVED** [review]

**Review by ViktoriiaLebedeva — CHANGES_REQUESTED** [review]

**Review by ViktoriiaLebedeva — APPROVED** [review]

**Review by IgorKey — APPROVED** [review]

**Pull request merged**